isthmus/src/test/java/io/substrait/isthmus/CustomFunctionTest.java

```diff
@@ -99,6 +113,51 @@ public RelDataType toCalcite(Type.UserDefined type) {
           null,
           SqlFunctionCategory.USER_DEFINED_FUNCTION);
 
+  static final SqlFunction customScalarAnyFn =
+      new SqlFunction(
+          "custom_scalar_any",
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.explicit(SqlTypeName.VARCHAR),
+          null,
+          null,
+          SqlFunctionCategory.USER_DEFINED_FUNCTION);
+
+  static final SqlFunction customScalarListAnyFn =
+      new SqlFunction(
+          "custom_scalar_listany",
+          SqlKind.OTHER_FUNCTION,
+          ReturnTypes.explicit(varcharArrayType),
```

Comment:
`custom_scalar_listany` is defined as returning a `LIST<any1>`. The return type here doesn't match that.

If I update the `R.STRING` types to be `R.I64` types in the `customScalarListAnyRoundtrip` the test fails.

Reply:
We have a problem if I try to make the type based on `SqlTypeName.ANY`:

```
java.lang.UnsupportedOperationException: Unable to convert the type ANY
	at io.substrait.isthmus.TypeConverter.toSubstrait(TypeConverter.java:141)
```

To add ANY in the TypeConverter we'll need to create an Any `io.substrait.type.Type`. Any concerns? Should I do within this PR?

Reply:
Ah. To clarify, I think we need a better return type inference here. Something like [ReturnTypes.ARG0](https://github.com/apache/calcite/blob/31d66e797d3ca66d154262b15ee35876804df963/core/src/main/java/org/apache/calcite/sql/type/ReturnTypes.java#L155-L160) but that inspects the array element type (which I think may exists within Calcite somewhere).